id: PYSEC-0000-CVE-2024-27351
details: 'In Django 3.2 before 3.2.25, 4.2 before 4.2.11, and 5.0 before 5.0.3, the
  django.utils.text.Truncator.words() method (with html=True) and the truncatewords_html
  template filter are subject to a potential regular expression denial-of-service
  attack via a crafted string. NOTE: this issue exists because of an incomplete fix
  for CVE-2019-14232 and CVE-2023-43665.'
affected:
- package:
    name: django
    ecosystem: PyPI
    purl: pkg:pypi/django
  ranges:
  - type: ECOSYSTEM
    events:
    - introduced: "3.2"
    - fixed: 3.2.25
    - introduced: "4.2"
    - fixed: 4.2.11
    - introduced: "5.0"
    - fixed: 5.0.3
references:
- type: WEB
  url: https://groups.google.com/forum/#%21forum/django-announce
- type: WEB
  url: https://docs.djangoproject.com/en/5.0/releases/security/
- type: ARTICLE
  url: https://www.djangoproject.com/weblog/2024/mar/04/security-releases/
aliases:
- CVE-2024-27351
modified: "2024-03-15T20:15:00Z"
published: "2024-03-15T20:15:00Z"
